public/app/features/alerting/unified/integration/LegacyAlertsDeprecationNotice.tsx

```diff
@@ -0,0 +1,92 @@
+import { isEmpty } from 'lodash';
+import React from 'react';
+import { useToggle } from 'react-use';
+
+import { config } from '@grafana/runtime';
+import { Dashboard, Panel, RowPanel } from '@grafana/schema';
+import { Alert, Collapse, Column, InteractiveTable, TextLink } from '@grafana/ui';
+
+import { makePanelLink } from '../utils/misc';
+
+interface DeprecationNoticeProps {
+  dashboard: Dashboard;
+}
+
+const usingLegacyAlerting = !config.unifiedAlertingEnabled;
+
+export default function LegacyAlertsDeprecationNotice({ dashboard }: DeprecationNoticeProps) {
+  // if the user is still using legacy alerting we don't need to show any notice at all – they will probably keep using legacy alerting and do not intend to upgrade.
+  if (usingLegacyAlerting) {
+    return null;
+  }
+
+  const panelsWithLegacyAlerts = getLegacyAlertPanelsFromDashboard(dashboard);
+
+  // don't show anything when the user has no legacy alerts defined
+  const hasLegacyAlerts = !isEmpty(panelsWithLegacyAlerts);
+  if (!hasLegacyAlerts) {
+    return null;
+  }
+
+  return dashboard.uid ? <LegacyAlertsWarning dashboardUid={dashboard.uid} panels={panelsWithLegacyAlerts} /> : null;
```

Comment:
Can `dashboard.uid` be undefined?

Reply:
Yep, that's how it seems to be typed – I suspect the same model is used for existing dashboard as for dashboards that are being created and don't have a UID from the database yet.